src/requireAuth.ts

```diff
@@ -0,0 +1,84 @@
+import { GoogleAuth, GoogleAuthOptions } from "google-auth-library";
+import * as clc from "cli-color";
+
+import * as api from "./api";
+import * as configstore from "./configstore";
+import { FirebaseError } from "./error";
+import * as logger from "./logger";
+import * as utils from "./utils";
+import * as scopes from "./scopes";
+
+const AUTH_ERROR_MESSAGE = `Command requires authentication, please run ${clc.bold(
+  "firebase login"
+)}`;
+
+let authClient: GoogleAuth | undefined;
+
+/**
+ * Returns the auth client.
+ * @param config options for the auth client.
+ */
+async function getAuthClient(config: GoogleAuthOptions): Promise<GoogleAuth> {
+  if (authClient) {
+    return authClient;
+  }
+
+  authClient = new GoogleAuth(config);
+  return authClient;
+}
+
+/**
+ * Retrieves and sets the access token for the current user.
+ * @param options CLI options.
+ * @param authScopes scopes to be obtained.
+ */
+async function autoAuth(options: any, authScopes: string[]): Promise<void> {
+  const client = await getAuthClient({ scopes: authScopes, projectId: options.project });
+  const token = await client.getAccessToken();
+  api.setAccessToken(token);
+}
+
+/**
+ * Ensures that there is an authenticated user.
+ * @param options CLI options.
+ */
+export async function requireAuth(options: any): Promise<void> {
+  api.setScopes([scopes.CLOUD_PLATFORM, scopes.FIREBASE_PLATFORM]);
+  options.authScopes = api.getScopes();
+
+  const tokens = configstore.get("tokens");
+  const user = configstore.get("user");
+
+  let tokenOpt = utils.getInheritedOption(options, "token");
+  if (tokenOpt) {
+    logger.debug("> authorizing via --token option");
+  } else if (process.env.FIREBASE_TOKEN) {
+    logger.debug("> authorizing via FIREBASE_TOKEN environment variable");
+  } else if (user) {
+    logger.debug("> authorizing via signed-in user");
+  } else {
+    return autoAuth(options, options.authScopes);
+  }
+
+  tokenOpt = tokenOpt || process.env.FIREBASE_TOKEN;
+
+  if (tokenOpt) {
+    api.setRefreshToken(tokenOpt);
+    return;
+  }
+
+  if (!user || !tokens) {
+    if (configstore.get("session")) {
+      throw new FirebaseError(
+        `This version of Firebase CLI requires reauthentication.
+
+Please run ${clc.bold("firebase login")} to regain access.`
+      );
+    }
```

Comment:
I'm pretty sure this can be removed. It's a super duper old thing it's checking for and I think it ends up throwing false positives instead of useful errors sometimes.

Reply:
Done.